tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/HasBeenUpdatedCounter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import com.codahale.metrics.Counter;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class HasBeenUpdatedCounter extends Counter {
+    private volatile boolean hasBeenUpdated = false;
```

Comment:
Is volatile necessary here? If a metric is updated prior to the reporting thread executing, happens-before relationship is fine, and parallel first use with the reporter can be reported with the next batch.

Reply:
Fair point, it does seem like a very low downside to race a call to `.inc()` with `.hasBeenUpdated()`!